url9a01f43c15_poragchowdhury_securitygame_tgz-pJ8-securitygame_GameMasterJ8/src/securitygame/Parser.java

```diff
@@ -0,0 +1,264 @@
+package securitygame;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+public class Parser
+{
+	
+	/**
+	 * Given a .graph file, a network is generated following a predetermined format
+	 * 
+	 * @param filename
+	 * @return a Network object based on the .graph file given
+	 */
+	public static Network parseGraph(String filename)
+	{
+		boolean deleteJunkFile = false;
+		try 
+		{
+			File gFile = new File(filename);
+			if(!gFile.exists()){
+				gFile.createNewFile();
+				deleteJunkFile = true;
+			}
+			CSVParser parser = CSVParser.parse(gFile, StandardCharsets.US_ASCII, CSVFormat.DEFAULT);
+			CSVParser parseRecords= CSVParser.parse(gFile, StandardCharsets.US_ASCII, CSVFormat.DEFAULT);
+			int neighborsCounter = 0;
+			int numNodes = parseRecords.getRecords().size()/2;
+            Network network = new Network(0,numNodes);
+            network.setName(filename.substring(0,filename.indexOf(".")));
+			boolean flag = false;
+			for (CSVRecord csvRecord : parser) 
+			{
+				Iterator<String> itr = csvRecord.iterator();
+				if((neighborsCounter<numNodes) && flag==false)
+				{
+					Node node = network.getNode(neighborsCounter);
+					while(itr.hasNext())
+					{
+						int x = Integer.parseInt(itr.next());
+						if(x >= 0){
+							Node neighbor = network.getNode(x);
+							node.addNeighbor(neighbor);
+						}
+					}
+					if(neighborsCounter==numNodes-1)
+					{
+						flag = true;
+						neighborsCounter=0;
+					}
+					else
+						neighborsCounter++;
+				}
+				else if(flag && (neighborsCounter<numNodes))
+				{
+					Node node = network.getNode(neighborsCounter);
+					while(itr.hasNext())
+					{
+						int x  = Integer.parseInt(itr.next());
+						node.setPv(x);
+						int y = Integer.parseInt(itr.next());
+						node.setSv(y);
+						int z = Integer.parseInt(itr.next());
+						node.setHoneyPot(z);
+					}
+					neighborsCounter++;
+				}
+			}
+			if(deleteJunkFile)
+				gFile.delete();
+			return network;
+		}
+		catch (IOException e) 
+		{
+			e.printStackTrace();
+		}
+		return null;
+	}
+    /**
+	 * Parses attacker's .history file and calculates and returns the attacker's network
+	 * @param attackerName
+	 * @param defenderName
+	 * @param graphName
+	 * @return attacker's visible graph
+	 */
+	public static Network parseAttackerHistory(String attackerName, String defenderName, String graphName)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 69 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.